src/coreclr/src/vm/method.cpp

```diff
@@ -5254,6 +5254,7 @@ void NDirectMethodDesc::InterlockedSetNDirectFlags(WORD wFlags)
 }
 
 #ifndef CROSSGEN_COMPILE
+#ifndef TARGET_UNIX
 FARPROC NDirectMethodDesc::FindEntryPointWithMangling(NATIVE_LIBRARY_HANDLE hMod, PTR_CUTF8 entryPointName) const
```

Comment:
I would add a comment as to why this function isn't supported on non-Windows.

Reply:
It's not so much not supported as unused. There's no reason to compile it for non-Windows since we never use it and the work it does isn't needed.

Reply:
Okay, then I would let the linker remove it instead of littering more #ifdefs.

Reply:
I see these ifdefs as a form of documentation. They say and enforce that this function is used on Windows only.

Reply:
Use is easy to ascertain from searching the code. The question I am asking is the "why". I don't find ifdefs to be much documentation any more than the `grep` or `findstr` tool since both will tell me if they are used. The "why" the function isn't needed or used is what my original statement was. ifdef'ing a function out merely obscures the reason and inevitably breeds doubt in line with "will or should this function work off Windows?". In this case, yes it will, it is simply ifdef'd because it isn't used.

I guess this is done so often it is a standard at this point, but the why is always more important than is it used in my opinion.